agent/main.go

```diff
@@ -119,7 +119,9 @@ func main() {
 	group.Go(func() error {
 		defer func() {
 			log.G(shimCtx).Info("stopping ttrpc server")
-			if err := server.Shutdown(shimCtx); err != nil {
+			// use context.Background() instead of the canceled shimCtx, which should allow shutdown to
+			// flush any pending responses before actually stopping the server
+			if err := server.Shutdown(context.Background()); err != nil {
```

Comment:
The [docs mention](https://golang.org/pkg/net/http/#Server.Shutdown) that `Shutdown` will: `... waiting indefinitely for connections to return to idle and then shut down`. Should we use context with timeout here?

Reply:
(This is a ttrpc server, not http, [but the same idea does apply](https://github.com/containerd/ttrpc/blob/master/server.go#L131))

I thought about that, but right now if we got blocked indefinitely on waiting for connections to become idle, then we'd just hit the shutdown timeout in the runtime shim and forcibly kill the VM. I think I'm okay with that because force-closing the server will often cause other errors to be returned to the runtime shim, which can result in the VM being forcibly closed anyways (i.e. `ttrpc: closed` errors to `agentClient.Shutdown`). Also, adding another timeout to a different part of the shutdown process is going to complicate it even further, which I'd really like to avoid unless we are sure we have to.